src/sql/workbench/parts/objectExplorer/browser/treeUpdateUtils.ts

```diff
@@ -151,57 +151,49 @@ export class TreeUpdateUtils {
-		tree: ITree): Promise<ConnectionProfile> {
-		return new Promise<ConnectionProfile>((resolve, reject) => {
-			if (!connectionManagementService.isProfileConnected(connection)) {
-				// don't try to reconnect if currently connecting
-				if (connectionManagementService.isProfileConnecting(connection)) {
-					resolve(undefined);
+		tree: ITree): Promise<ConnectionProfile | undefined> {
+		if (!connectionManagementService.isProfileConnected(connection)) {
+			// don't try to reconnect if currently connecting
+			if (connectionManagementService.isProfileConnecting(connection)) {
+				return undefined;
 
-					// else if we aren't connected or connecting then try to connect
-				} else {
-					let callbacks: IConnectionCallbacks = undefined;
-					if (tree) {
-						// Show the spinner in OE by adding the 'loading' trait to the connection, and set up callbacks to hide the spinner
-						tree.addTraits('loading', [connection]);
-						let rejectOrCancelCallback = () => {
-							tree.collapse(connection);
-							tree.removeTraits('loading', [connection]);
-						};
-						callbacks = {
-							onConnectStart: undefined,
-							onConnectReject: rejectOrCancelCallback,
-							onConnectSuccess: () => tree.removeTraits('loading', [connection]),
-							onDisconnect: undefined,
-							onConnectCanceled: rejectOrCancelCallback,
-						};
-					}
-					connectionManagementService.connect(connection, undefined, options, callbacks).then(result => {
-						if (result.connected) {
-							let existingConnection = connectionManagementService.findExistingConnection(connection);
-							resolve(existingConnection);
-						} else {
-							reject('connection failed');
-						}
-					}, connectionError => {
-						reject(connectionError);
-					});
-				}
+				// else if we aren't connected or connecting then try to connect
 			} else {
-				let existingConnection = connectionManagementService.findExistingConnection(connection);
-				if (options && options.showDashboard) {
-					connectionManagementService.showDashboard(connection).then((value) => {
-						resolve(existingConnection);
-					});
+				let callbacks: IConnectionCallbacks = undefined;
+				if (tree) {
+					// Show the spinner in OE by adding the 'loading' trait to the connection, and set up callbacks to hide the spinner
+					tree.addTraits('loading', [connection]);
+					let rejectOrCancelCallback = () => {
+						tree.collapse(connection);
+						tree.removeTraits('loading', [connection]);
+					};
+					callbacks = {
+						onConnectStart: undefined,
+						onConnectReject: rejectOrCancelCallback,
+						onConnectSuccess: () => tree.removeTraits('loading', [connection]),
+						onDisconnect: undefined,
+						onConnectCanceled: rejectOrCancelCallback,
+					};
+				}
+				const result = await connectionManagementService.connect(connection, undefined, options, callbacks);
+				if (result.connected) {
+					let existingConnection = connectionManagementService.findExistingConnection(connection);
```

Comment:
Couldn't we do `return result.connectionProfile as ConnectionProfile` here?

Reply:
I wouldn't trust that that is how it works.